### DISCUSSION TITLE
planar transmission line examples

### DISCUSSION THREAD

**Discussion by PinkMushroom** [NanoComp/meep#2493] [Q&A] [answered]
To get familiar with meep I'm going to build a basic microstrip model for simulation.

1 does anyone know of some code to give me a starting point ?
2 any gotchas i should be aware of in setting up that type of model (e.g. should I use Mur or ABC ?) ?
3 meep really seems to be oriented around photonics, and yet also seems to be a general FDTD implementation . There isn't any reason why simulating planar transmission line structures would be a problem, is there ?

Thank you!

**Comment by tmteryz** [comment, marked as answer]
In most cases you should use [PML](https://meep.readthedocs.io/en/latest/Perfectly_Matched_Layer/) in meep.
for the simulation of  microstrip ,you can  refer to this tutorial [https://meep.readthedocs.io/en/latest/Python_Tutorials/Eigenmode_Source/](url). 
but if the microstrip  is made of PEC or lossy metal,  maybe you need to develop the function of discrete port by yourself(see #2457).

**Reply by PinkMushroom** [reply]
Yes the microstrip is made of PEC or lossy metal.

Reading #2457 tells me meep has nothing suitable for providing a source for microwave planar transmission lines type problems.

meep is just very, very photonics-centric.

Anyone who wants to use it for microwave style problems, like me, is going to have a lot of work to do.

**Reply by smartalecH** [reply]
> Anyone who wants to use it for microwave style problems, like me, is going to have a lot of work to do.

Why? It's rather straightforward to inject an RF mode from an appropriate mode solver.

**Reply by manujmg** [reply]
Any update on this issue?
@smartalecH what do you mean by injecting an RF mode from an appropriate mode solver?

You would need to create a custom source function to inject an RF signal carrying the mode you are interested but also a mode decomposition/solver custom function. Right?

**Reply by smartalecH** [reply]
> Any update on this issue?

Nope.

> You would need to create a custom source function

Meep already has that. See [here](https://github.com/NanoComp/meep/blob/295fa633cefb016a8ffd52d98e393ae7781e2bc7/python/source.py#L774C5-L774C22). You just need to get the right mode profiles from an external solver.

>  but also a mode decomposition/solver custom function

Sure you could do this in python pretty easily though. Just a mode overlap with the recorded transverse E and H fields.

**Reply by manujmg** [reply]
@smartalecH But, let's assume we have a 3D simulation with a microstrip along the x axis and a gaussian source in the Y-Z  (mp.Ez component) plane (much wider and longer than the microstrip dimensions) simulating an infinitely waveguide connected to the microstrip.

By running a standard simulation and getting the fields at some point in the microstrip (e.g. accumulating the appropriate field Fourier transforms in a flux monitor and getting those with get_fluxes)... couldn't I use those to do the mode decomposition, get the mode profiles and use those profiles with the GaussianBeam3DSource so I excite the microstrip properly?

My real question here would be... how would you extract the S11 Parameter in an scenario like that....

**Reply by smartalecH** [reply]
Sounds like we're diverging a bit from the main thread. Maybe open a new discussion (with a figure describing your setup, if you can).

**Reply by manujmg** [reply]
@smartalecH there you have: #2743
